test/test_namedtensor.py

```diff
@@ -23,10 +24,58 @@ def flatten(lst):
 Function = namedtuple('TestCase', ['name', 'lambd'])
 
 
+def parse_compressed_namedshape(string):
+    # This is a metalanguage for describing a shape of a tensor compactly.
+    # 'N:3,C:2' -> size = [3, 2], names: ['N', 'C']
+    # 'None:3,None:2' -> size = [3, 2], names: ['None', 'None']
+    # '3,2' -> size = [3, 2], names=None passed to ctor.
+    def parse_name(maybe_name):
+        maybe_name = maybe_name.strip()
+        if maybe_name == 'None':
+            return None
+        return maybe_name
+
+    string = string.strip()
+
+    # '3, 2' -> size = [3, 2], None names.
+    if len(string) > 0 and ':' not in string:
+        return None, [int(size) for size in string.split(',')]
+
+    dims = string.split(',')
+    tuples = [dim.split(':') for dim in dims]
+    return zip(*[(parse_name(name), int(size)) for name, size in tuples])
+
+
+def create(namedshape, factory=torch.randn):
+    # namedshape: str
+    names, shape = parse_compressed_namedshape(namedshape)
+    return factory(shape, names=names)
+
+
+def out_fn(operator):
+    @functools.wraps(operator)
+    def fn(*inputs):
+        return operator(*inputs[1:], out=inputs[0])
+    return fn
+
+
 class TestNamedTensor(TestCase):
     def test_trivial(self):
         pass
 
+    def _test_name_inference(self, op, args=(), expected_names=(), device='cpu',
+                             maybe_raises_regex=None):
+        casted_args = [arg.to(device) if isinstance(arg, torch.Tensor) else arg
+                       for arg in args]
+        if maybe_raises_regex is not None:
+            with self.assertRaisesRegex(RuntimeError, maybe_raises_regex):
```

Comment:
i don't see this used... should we test some cases that fail (e.g. mis-matched names in a mm op should fail right?)

Reply:
Right now, the name inference rule for mm doesn't fail if names are mismatched, so that won't fail.

Reply:
is it supposed to fail?

Reply:
The decision right now is for it to not fail. Tensor[M, N] @ Tensor[O, P] returns Tensor[M, P] and it doesn't matter that N / O are "mismatched".

Reply:
ah now I understand, thanks